content/blog/testthat-3-0-0/index.Rmd

```diff
@@ -0,0 +1,113 @@
+
+-   `expect_error()`, `expect_warning()`, `expect_message()`, and `expect_condition()` now all use the same underlying logic: they capture the first condition that matches `class`/`regexp` and allow anything else to bubble up.
+    This behaviour pairs particularly well with upcoming changes to the pipe in [magrittr 2.0.0](/blog/2020/08/magrittr-2-0/).
+
+-   Messages are no longer automatically silenced.
+    You'll now need to use `suppressMessages()` to hide unimportant messages or `expect_messsage()` to catch important messages.
+
+-   `test_that()` now sets a number of options and env vars to make output as reproducible as possible.
+    Many of these options were previously set in various places (including `devtools::test()`, `test_dir()`, `test_file()`, and `verify_output()`) but they now been centralised in to `local_test_context()`.
+
+## Snapshot testing
+
+The goal of a unit test is to record the expected output of a function using code.
+This is a powerful technique because because not only does it ensure that code doesn't change unexpectedly, it also expresses the desired behaviour in a way that a human can understand.
+However, it's not always convenient to record the expected behaviour with code.
+Some challenges include:
+
+-   Text output that includes many characters like quotes and newlines that require special handling in a string.
+
+-   Output that is large, making it painful to define the reference output, and bloating the size of the test file and making it hard to navigate.
+
+-   Cases where you want to record a mix of printed output, messages, warnings, or errors.
+
+-   Binary formats like plots or images, which are very difficult to describe in code: i.e. the plot looks right, the error message is useful to a human, the print method uses colour effectively.
+
+For these situations, testthat provides an alternative mechanism: snapshot tests.
+Instead of using code to describe expected output, snapshot tests (also known as [golden tests](https://ro-che.info/articles/2017-12-04-golden-tests)) record results in a separate human readable file.
+Snapshot tests in testthat are inspired primarily by [Jest](https://jestjs.io/docs/en/snapshot-testing), thanks to a number of very useful discussions with Joe Cheng.
+
+If snapshot tests sound intriguing to you, please read the details in `vignette("snapshotting", package = "testthat").`
+
+## Other improvements
+
+-   testthat has preliminary support for running tests in parallel.
+    See details in `vignette("parallel", package = "testthat")`
+
+-   We now recommend using test fixtures rather than `setup()` and `teardown()` code.
+    See details in `vignette("test-fixtures", package = "testthat")`.
+
+-   We have revamped non-interactive reporter to a better job of displaying skipped tests and written a vignette, `vignette("skipping", package = "testthat")`, describing best practices (including how to test your own skip functions).
```

Comment:
```suggestion
-   We have revamped the non-interactive reporter to do a better job of displaying skipped tests, and have written a vignette, `vignette("skipping", package = "testthat")`, describing best practices (including how to test your own skip functions).
```

Reply:
Double check these edits, you're definitely missing an article, and the word "do," but I'm not sure if the non-interactive reporter is singular.

Reply:
It's technically plural, I think. I'll fix it.